src/github.com/matrix-org/dendrite/syncserver/storage/output_room_events_table.go

```diff
@@ -47,8 +47,10 @@ const selectRecentEventsSQL = "" +
 const selectMaxIDSQL = "" +
 	"SELECT MAX(id) FROM output_room_events"
 
+// In order for us to apply the state updates correctly, rows need to be ordered in the order they were received (id).
 const selectStateInRangeSQL = "" +
-	"SELECT event_json, add_state_ids, remove_state_ids FROM output_room_events WHERE id > $1 AND id < $2 AND add_state_ids IS NOT NULL OR remove_state_ids IS NOT NULL"
+	"SELECT event_json, add_state_ids, remove_state_ids FROM output_room_events WHERE id > $1 AND id < $2 AND " +
+	"add_state_ids IS NOT NULL OR remove_state_ids IS NOT NULL ORDER BY id ASC"
```

Comment:
You've got the job of putting this in the style guide :) I don't disagree with you at all to be fair, it does look nicer.

Done. https://github.com/matrix-org/dendrite/pull/71/commits/9fee01e5bf61bbd93be9e4945848c3b5f7679078